content/raw/communities/community_history.md

```diff
@@ -0,0 +1,83 @@
+
+Further information on those problems can be found in these documents:
+
+- [Waku V2 Usage for Community History](https://github.com/vacp2p/research/issues/83)
+- [Capturing messages order and dependency](https://github.com/vacp2p/research/issues/76)
+- [Access Control Management](https://github.com/vacp2p/research/issues/79)
+- [Store synchronization/consensus in Waku V2 store protocol](https://github.com/vacp2p/research/issues/80)
+- [Offline/long-term storage](https://github.com/vacp2p/research/issues/81)
+- [Resource throttling](https://github.com/vacp2p/research/issues/84)
+- [Edit/remove messages](https://github.com/vacp2p/research/issues/77)
+
+
+#### A TLDR; problem description (MVP)
+A quick summary of the problem space of the MVP:
+
+1. **Message Ordering** - Messages are ordered using sender generated timestamp + message hash. Assumes nodes have synchronized clocks and max. +- 20 seconds asynchrony. This is covered by [WAKU2-FTSTORE](https://rfc.vac.dev/spec/21/).
+2. **State synchronization/consensus** - Store nodes listen to network and store all communities messages. At least one node has to be online at any given time. Nodes can recover message gaps by performing time-based queries to other nodes. This is covered by [WAKU2-FTSTORE](https://rfc.vac.dev/spec/21/)
+3. **Discovery** - Static and provided by community owner.
+4. **Transport** - Waku2 (Gossipsub)
+5. **Offline/long-term storage** - Nodes should have enough storage to maintain entire history.
+
+<img style="max-width: 100%" src="https://camo.githubusercontent.com/acd1d1f15ca077f4a9386611800b65c8b4b44b5ac623e00c4ac6e8e0dc722f24/68747470733a2f2f692e696d6775722e636f6d2f6d364b71704b762e6a7067">
+
+### Use cases
+
+These are some rough notes that need to be turned into cockburn style notes:
+
+- As a user I want to be able to read messages of community channels prior to my membership
+- As a user I want to be able to read messages older than 30 days
+
+...
+
+### Functional requirements
+
+#### General
+
+- The community owner MUST run store nodes
+  - Store nodes MUST implement [WAKU2-STORE](https://rfc.vac.dev/spec/13/) and [WAKU2-FTSTORE](https://rfc.vac.dev/spec/21/)
+  - Store nodes MUST be configured with a static list of all store nodes
+  - Store nodes MUST have a synchronized clock with at most +- 20 seconds difference
+  - Store nodes MUST be participants of Gossipsub
+
+#### Creating communities
+
+- The community owner MUST specify a list of trusted store nodes and their addresses
+- The store node list MUST be public to community members
+- The Status client MUST add community store nodes to its known store node cluster
+
+#### Editing communities
+- The community owner MAY change the list of store nodes for the community
+  - The Status client MUST update its known store nodes
+  - The Status client MUST fetch history data when known store nodes were changed
```

Comment:
Is this true? I assume that, when store nodes have been changed, there's a possibility they might have more history data. So re-fetching at this point might be desired. Thoughts?

Reply:
Good question.. Maybe not a possibility, but I was thinking, maybe there's some nodes that received messages that other didn't and the owner then decides to put those more "complete" nodes as store nodes for his community.

But now that I think of it, that would mean that the other nodes that were set before live in their own network, being isolated in some way, therefore not having had access to other messages.

So yea, we can probably remove this requirement